packages/pigeon/lib/kotlin_generator.dart

```diff
@@ -679,7 +679,7 @@ String _castForceUnwrap(String value, TypeDeclaration type, Root root) {
     // longs in Pigeon with Kotlin.
     if (type.baseName == 'int') {
       final String castUnwrap = type.isNullable ? '?' : '';
-      return '$value.let { if (it is Int) it.toLong() else it as$castUnwrap Long }';
+      return '$value.let { if (it is Int) it.toLong() else it as Long$castUnwrap }';
     } else {
       return _cast(value,
           kotlinType: _kotlinTypeForDartType(type), safeCast: type.isNullable);
```

Comment:
Same comments here.

And since this is basically the same code as above, a helper for this special casing of int would be a good idea.